custom_components/waste_collection_schedule/waste_collection_schedule/source/adur_worthing_gov_uk.py

```diff
@@ -0,0 +1,80 @@
+import requests
+import bs4
+import re
+
+from datetime import datetime
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+
+TITLE = "Adur & Worthing Councils"
+DESCRIPTION = "Source for adur-worthing.gov.uk services for Adur & Worthing, UK."
+URL = "https://adur-worthing.gov.uk"
+TEST_CASES = {
+    "Test_001": {"postcode": "BN15 9UX", "address": "1 Western Road North"},
+    "Test_002": {"postcode": "BN43 5WE", "address": "6 Hebe Road"}
+}
+HEADERS = {
+    "user-agent": "Mozilla/5.0",
+}
+ICON_MAP = {
+    "Recycling": "mdi:recycle",
+    "Refuse": "mdi:trash-can",
+    "Garden": "mdi:leaf",
+}
+
+
+class Source:
+    def __init__(self, postcode, address):
+        self._postcode = postcode
+        self._address = address
+
+    def fetch(self):
+
+        if (self._postcode is None or self._address is None):
+            return []
+
+        s = requests.Session()
+
+        postcode_search_request = s.get(
+            f"https://www.adur-worthing.gov.uk/bin-day/?brlu-address-postcode={self._postcode}&return-url=/bin-day/&action=search",
+            headers=HEADERS
+        )
+        html_addresses = postcode_search_request.content
+        addresses = bs4.BeautifulSoup(html_addresses, "html.parser")
+        addresses_select = addresses.find('select', { 'id': 'brlu-selected-address' })
+
+        found_address = None
+        for address in addresses_select.find_all('option'):
+            if self._address in address.get_text():
+                found_address = address
+        
+        if found_address is None:
+            return []
```

Comment:
same as above, indicate a that a wrong/not-supported address was provided by raising an Exception